### DISCUSSION TITLE
How to properly set date(time) field ?

### DISCUSSION THREAD

**Discussion by vvoila** [adonisjs/core#4578] [Help]
I dont have this issue in v5, but in v6 is a bit confusing and i hope someone can give a clue as whats is going on.

So let's say in mysql database theres users table with fields dob as date and last_seen as datetime.

heres my model declaration:

```
export default class User extends BaseModel {
  @column.date()
  declare dob: DateTime | null

  @column.dateTime()
  declare lastSeen: DateTime | null
```

the (vine) validators:

```
export const createUserValidator = vine.compile(
  vine.object({
    dob: vine.date().nullable().optional(),
    lastSeen: vine.date().nullable().optional(),
  })
)
```

controllers, to store data:

```
async store({ request }: HttpContext) {
  const data = await request.validateUsing(createUserValidator)
  const result = await User.create(data)

  return result
}
```

in editor, got this warning in line 4: Types of property 'dob' are incompatible. Type 'Date | null | undefined' is not assignable to type 'DateTime<boolean> | null | undefined'. Type 'Date' is missing the following properties from type 'DateTime<boolean>': get, getPossibleOffsets, isValid, invalidReason, and 71 more.ts(2345)

as for sending post to the endpoint, with parameter dob e.g. "2000-10-10" i got the following error message: "Invalid value for \"User.dob\". The value must be an instance of \"luxon.DateTime\""

any idea?

**Comment by sinh117801** [comment]
hope this help for dob

```typescript
// In your validator
import { DateTime } from 'luxon';

...
dob: vine.date().optional().transform(value => DateTime.fromJSDate(value))
...
```

**Comment by aikrom** [comment]
I guess we can look at this as a feature request or a bug. Since the framework uses `luxon.DateTime` to work with dates, the result of the validation (and transformation) is expected to be `luxon.DateTime`.

Option 1: We can create custom rule to transform to `luxon.DateTime`

``` ts
const schema = vine.object({
    date: vine.date().luxon(),
})
```

Option 2: We can create custom schema type

``` ts
const schema = vine.object({
    date: vine.dateTime(), // or vine.luxonDateTime()
})
```
Option 3: We can override the default date schema type (but I'm not sure if this is possible)

``` ts
const schema = vine.object({
    date: vine.date(), // convert value to DateTime
})
```

If someone from the core team gives feedback and recommendations, we may be able to make some contribution.

**Comment by 25Das** [comment]
My solution is use transform
```
@column.dateTime()
declare date: DateTime | null
```
```
const schema = vine.object({
    date: vine.date().transform((value) => value && DateTime.fromJSDate(value).nullable(),
})
```
